### PR TITLE
Float: Introduce floating point api lt_quiet

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* The lt_quiet(less than quiet) api(s) introduced.
* Add test cases for half, single, double and quad floating point.
* Add lt to interface.

### CODE REVIEW

**Comment by jordancarlin** [comment]
There is a lot of duplicated logic with the float_is_lt function. I wonder if float_is_lt_quiet should call float_is_lt and just change the flag. Or maybe even change float_is_lt to take another boolean input (is_quiet) and have it return either the normal or quiet version based on that input.

**Comment by Incarnation-p-lee** [comment]
I see, let me extract a func lt without any exception for this.

File: lib/float/lt_quiet.sail [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+$ifndef _FLOAT_LT_QUIET
+$define _FLOAT_LT_QUIET
+
+$include <float/common.sail>
+$include <float/zero.sail>
+$include <float/nan.sail>
+$include <float/arith_internal.sail>
+
+val      float_is_lt_quiet : fp_bits_x2 -> fp_bool_and_flags
+function float_is_lt_quiet ((op_0, op_1)) = {
```
Comment (arichardson) [inline lib/float/lt_quiet.sail:39]: Unrelated to this change, but can we avoid the need for tuples when calling these functions?
Maybe @Alasdair has a suggestion how to write the type constraint that both arguments must be the same bitwidth?

Comment (Alasdair) [reply lib/float/lt_quiet.sail:39]: The only way to do this is to add the constraint to the function, i.e.
```
type is_fp_bits('n) = 'n in {16, 32, 64, 128}
val float_is_lt_quiet : forall 'n, is_fp_bits('n). (bits('n), bits('n)) -> bool
```
there isn't any way to add a synonym that covers multiple function arguments simultaneously

**Comment by Incarnation-p-lee** [comment]
Kindly ping, ;)!

**Pull request merged**